samza-core/src/main/scala/org/apache/samza/config/StreamConfig.scala

```diff
@@ -145,10 +145,14 @@ class StreamConfig(config: Config) extends ScalaMapConfig(config) with Logging {
     * Gets the physical name for the specified streamId.
     *
     * @param streamId             the identifier for the stream in the config.
-    * @param defaultPhysicalName  the default to use if the physical name is missing.
     * @return                     the physical identifier for the stream or the default if it is undefined.
     */
-  def getPhysicalName(streamId: String, defaultPhysicalName: String) = {
+  def getPhysicalName(streamId: String) = {
+    // Add prefix to default so the physical name is unique per job
+    val defaultPhysicalName = String.format("%s-%s-%s",
```

Comment:
One of the considerations behind picking a default physical name was to make it the same as the logical name, so that for systems like Kafka etc. you could make your logical ID == default physical name == PageViewEvent. From that perspective, does it make sense to put the job name and job id in the logical stream ID instead?

Also, let's check if this default naming convention is part of public documentation for streams.* configs.